native/src/modules/app/createNavigateToCategory.js

```diff
@@ -2,38 +2,41 @@
 
 import type { Dispatch } from 'redux'
 import type { FetchCategoryActionType, StoreActionType } from './StoreActionType'
-import type { NavigationStackProp } from 'react-navigation-stack'
 import { generateKey } from './generateRouteKey'
+import type {
+  CategoriesRouteType,
+  DashboardRouteType,
+  NavigationPropType,
+  RoutesType
+} from './components/NavigationTypes'
 
 export type NavigateToCategoryParamsType = {|
   cityCode: string, language: string, path: string, key?: string, forceRefresh?: boolean
 |}
 
-export default (
-  routeName: 'Categories' | 'Dashboard',
+const createNavigateToCategory = <T: RoutesType>(
+  routeName: CategoriesRouteType | DashboardRouteType,
   dispatch: Dispatch<StoreActionType>,
-  navigation: NavigationStackProp<*>
+  navigation: NavigationPropType<T>
 ) => ({ cityCode, language, path, key = generateKey(), forceRefresh = false }: NavigateToCategoryParamsType) => {
-  navigation.navigate({
-    routeName,
-    params: {
-      onRouteClose: () => dispatch({ type: 'CLEAR_CATEGORY', params: { key } }),
```

Comment:
Is it oke that this is removed here?

Reply:
Yes, this functionality is now handled in the payload provider as well.